subsys/audio_module/audio_module.h

```diff
@@ -0,0 +1,463 @@
+	 */
+	int (*close)(struct audio_module_handle_private *handle);
+
+	/**
+	 * @brief Function to configure a module.
+	 *
+	 * @param handle[in/out]     The handle to the module instance
+	 * @param configuration[in]  Pointer to the module's configuration to set
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*configuration_set)(struct audio_module_handle_private *handle,
+				 struct audio_module_configuration *configuration);
+
+	/**
+	 * @brief Function to get the configuration of a module.
+	 *
+	 * @param handle[in]          The handle to the module instance
+	 * @param configuration[out]  Pointer to the module's current configuration
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*configuration_get)(struct audio_module_handle_private *handle,
+				 struct audio_module_configuration *configuration);
+
+	/**
+	 * @brief Start a module.
+	 *
+	 * @param handle[in/out]  The handle for the module to start
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*start)(struct audio_module_handle_private *handle);
+
+	/**
+	 * @brief Stop a module.
+	 *
+	 * @param handle[in/out]  The handle for the module to stop
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*stop)(struct audio_module_handle_private *handle);
+
+	/**
+	 * @brief The core data processing function for the module. Can be either an
+	 *		  input, output or processor module type.
+	 *
+	 * @param handle[in/out]  The handle to the module instance
+	 * @param block_rx[in]    Pointer to the input audio block or NULL for an input module
+	 * @param block_tx[out]   Pointer to the output audio block or NULL for an output module
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*data_process)(struct audio_module_handle_private *handle, struct audio_data *block_rx,
+			    struct audio_data *block_tx);
+};
+
+/**
+ * @brief A module's minimum description.
+ */
+struct audio_module_description {
+	/* The module base name */
+	char *name;
+
+	/* The module type */
+	enum audio_module_type type;
+
+	/* A pointer to the functions in the module */
+	const struct audio_module_functions *functions;
+};
+
+/**
+ * @brief Module's thread configuration structure.
+ */
+struct audio_module_thread_configuration {
+	/* Thread stack */
+	k_thread_stack_t *stack;
+
+	/* Thread stack size */
+	size_t stack_size;
+
+	/* Thread priority */
+	int priority;
+
+	/* A pointer to a module's data receiver FIFO, can be NULL */
+	struct data_fifo *msg_rx;
+
+	/* A pointer to a module's data transmitter FIFO, can be NULL */
+	struct data_fifo *msg_tx;
+
+	/* A pointer to the data buffer slab, can be NULL */
+	struct k_mem_slab *data_slab;
+
+	/* Size of each memory block in bytes that will be
+	 * taken from the data buffer slab or 0
+	 */
+	size_t data_size;
+};
+
+/**
+ * @brief Module's generic set-up structure.
+ */
+struct audio_module_parameters {
+	/* The module's private description */
+	struct audio_module_description *description;
```

Comment:
If this is private, should we have "priv" in the name or perhaps 
```suggestion
	struct audio_module_description *description_;
``` ?

Reply:
I don't think this actually needs to be private, will update comment.